In [1]:
import pandas as pd
import time, datetime
import multiprocessing as mp
from datetime import datetime as dt

# Load data

In [6]:
bad_cols = [0,2,6,15,16,17,18,19,20,21,22,24,25,26] # unnecessary columns from example_data notebook
cols = list(range(0,26)) # list of all columns
good_cols = [i for i in cols if i not in bad_cols] # create list of useful columns
good_cols = [str(x) for x in good_cols] # convert to string
good_cols

['1', '3', '4', '5', '7', '8', '9', '10', '11', '12', '13', '14', '23']

In [7]:
start = time.time()
# read data in chunks of 100k rows at a time
chunk = pd.read_csv('feb2021.csv.zip', 
                    chunksize=100000, 
                    dtype={'3': 'str'}) # specify index column
# iter_csv = pd.read_csv('file.csv', iterator=True, chunksize=1000)
# df = pd.concat([chunk[chunk['field'] > constant] for chunk in iter_csv])
df = pd.concat(chunk) # concat chunks together
end = time.time()
print("Read csv with chunks: ",(end-start),"sec")

Read csv with chunks:  7.836688041687012 sec


In [8]:
df

1                      3  \
0       20210201000000.0               tass.com   
1       20210201000000.0  observer-reporter.com   
2       20210201000000.0        jordantimes.com   
3       20210201000000.0             apnews.com   
4       20210201000000.0               wdiy.org   
...                  ...                    ...   
151925    20210201234500           whiznews.com   
151926    20210201234500    trinidadexpress.com   
151927    20210201234500             iheart.com   
151928    20210201234500             iheart.com   
151929    20210201234500     couriermail.com.au   

                                                        4    5  \
0                    https://tass.com/emergencies/1250941  NaN   
1       https://observer-reporter.com/publications/shl...  NaN   
2       http://jordantimes.com/news/local/ujs-css-tops...  NaN   
3       https://apnews.com/1ce7ac194ff97c882f2fe79eebb...  NaN   
4       https://www.wdiy.org/post/hilton-valentine-fou...  NaN   
...                                                   ...  ...   
151925  https://whiznews.com/2021/02/01/liverpool-fill...  NaN   
151926  https://trinidadexpress.com/opinion/letters/la...  NaN   
151927  https://wfmf.iheart.com/content/2021-02-01-mil...  NaN   
151928  https://magic1079.iheart.com/content/2021-02-0...  NaN   
151929  https://www.couriermail.com.au/entertainment/t...  NaN   

                                                        7  \
0       TAX_FNCACT;TAX_FNCACT_CHILDREN;ARREST;CRISISLE...   
1       GENERAL_HEALTH;HEALTH_PANDEMIC;WB_635_PUBLIC_H...   
2       EDUCATION;SOC_POINTSOFINTEREST;SOC_POINTSOFINT...   
3       TAX_DISEASE;TAX_DISEASE_CANCER;WB_1406_DISEASE...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY;TAX_ETHNICITY_ASHANTI;KILL;MANMA...   
151927  TAX_FNCACT;TAX_FNCACT_SINGER;TAX_ETHNICITY;TAX...   
151928  PUBLIC_TRANSPORT;TRIAL;TAX_WORLDFISH;TAX_WORLD...   
151929  TAX_ECON_PRICE;MEDIA_SOCIAL;TAX_FNCACT;TAX_FNC...   

                                                        8  \
0       SOC_POINTSOFINTEREST_POLICE_STATIONS,314;TAX_F...   
1       MANMADE_DISASTER_IMPLIED,1548;MANMADE_DISASTER...   
2       TAX_FNCACT_DIRECTOR,1343;USPEC_POLICY1,769;USP...   
3       MANMADE_DISASTER_RADIOACTIVE,975;ARMEDCONFLICT...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY_ASHANTI,12;MANMADE_DISASTER_IMPL...   
151927  MEDIA_SOCIAL,891;TAX_FNCACT_SINGER,98;TAX_FNCA...   
151928  TRIAL,77;TAX_WORLDFISH_FISH,523;UNGP_FORESTS_R...   
151929  MEDIA_MSM,1795;MEDIA_SOCIAL,594;TAX_ECON_PRICE...   

                                                        9  \
0       4#Moscow, Moskva, Russia#RS#RS48#55.7522#37.61...   
1                  1#Lebanon#LE#LE#33.833333#35.833333#LE   
2       4#Amman, (Jo11), Jordan#JO#JO11#31.95#35.9333#...   
3       2#California, United States#US#USCA#36.17#-119...   
4       4#North Shields, North Tyneside, United Kingdo...   
...                                                   ...   
151925  1#Ecuador#EC#EC#-2#-77.5#EC;1#Japan#JA#JA#36#1...   
151926                                                NaN   
151927                                                NaN   
151928                                                NaN   
151929                                                NaN   

                                                       10  \
0       4#Moscow, Moskva, Russia#RS#RS48#25106#55.7522...   
1            1#Lebanon#LE#LE##33.833333#35.833333#LE#2280   
2       1#Jordan#JO#JO##31#36#JO#13;1#Jordan#JO#JO##31...   
3       3#Los Angeles, California, United States#US#US...   
4       5#Northumberland, Northumberland, United Kingd...   
...                                                 

In [9]:
df[df['1'].isnull()]

1    3    4                                                  5    7  \
16      NaN  NaN  NaN                                                NaN  NaN   
5222    NaN  NaN  NaN                                                NaN  NaN   
13382   NaN  NaN  NaN                                                NaN  NaN   
13383   NaN  NaN  NaN                                                NaN  NaN   
13384   NaN  NaN  NaN                                                NaN  NaN   
...     ...  ...  ...                                                ...  ...   
147202  NaN  NaN  NaN                                                NaN  NaN   
147203  NaN  NaN  NaN                                                NaN  NaN   
147204  NaN  NaN  NaN                                                NaN  NaN   
147205  NaN  NaN  NaN                                                NaN  NaN   
147206  NaN  NaN  NaN  <PAGE_LINKS>https://katiecavuto.com/;https://s...  NaN   

          8    9   10   11   12   13   14   23  
16      NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
5222    NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
13382   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
13383   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
13384   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
...     ...  ...  ...  ...  ...  ...  ...  ...  
147202  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
147203  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
147204  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
147205  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
147206  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[230 rows x 13 columns]

In [10]:
df[df['3'].isnull()]

1    3    4  \
16                                                    NaN  NaN  NaN   
5222                                                  NaN  NaN  NaN   
13381   Content By Slug,5376;HeartRadio Popular Podcas...  NaN  NaN   
13382                                                 NaN  NaN  NaN   
13383                                                 NaN  NaN  NaN   
...                                                   ...  ...  ...   
147202                                                NaN  NaN  NaN   
147203                                                NaN  NaN  NaN   
147204                                                NaN  NaN  NaN   
147205                                                NaN  NaN  NaN   
147206                                                NaN  NaN  NaN   

                                                        5    7    8    9   10  \
16                                                    NaN  NaN  NaN  NaN  NaN   
5222                                                  NaN  NaN  NaN  NaN  NaN   
13381                                                 NaN  NaN  NaN  NaN  NaN   
13382                                                 NaN  NaN  NaN  NaN  NaN   
13383                                                 NaN  NaN  NaN  NaN  NaN   
...                                                   ...  ...  ...  ...  ...   
147202                                                NaN  NaN  NaN  NaN  NaN   
147203                                                NaN  NaN  NaN  NaN  NaN   
147204                                                NaN  NaN  NaN  NaN  NaN   
147205                                                NaN  NaN  NaN  NaN  NaN   
147206  <PAGE_LINKS>https://katiecavuto.com/;https://s...  NaN  NaN  NaN  NaN   

         11   12   13   14   23  
16      NaN  NaN  NaN  NaN  NaN  
5222    NaN  NaN  NaN  NaN  NaN  
13381   NaN  NaN  NaN  NaN  NaN  
13382   NaN  NaN  NaN  NaN  NaN  
13383   NaN  NaN  NaN  NaN  NaN  
...     ...  ...  ...  ...  ...  
147202  NaN  NaN  NaN  NaN  NaN  
147203  NaN  NaN  NaN  NaN  NaN  
147204  NaN  NaN  NaN  NaN  NaN  
147205  NaN  NaN  NaN  NaN  NaN  
147206  NaN  NaN  NaN  NaN  NaN  

[275 rows x 13 columns]

In [11]:
df['1'] = pd.to_numeric(df['1'],errors='coerce')
df = df.drop(df[df['1'].isnull()].index, axis=0)
df = df.drop(df[df['3'].isnull()].index, axis=0)
df

1                      3  \
0       2.021020e+13               tass.com   
1       2.021020e+13  observer-reporter.com   
2       2.021020e+13        jordantimes.com   
3       2.021020e+13             apnews.com   
4       2.021020e+13               wdiy.org   
...              ...                    ...   
151925  2.021020e+13           whiznews.com   
151926  2.021020e+13    trinidadexpress.com   
151927  2.021020e+13             iheart.com   
151928  2.021020e+13             iheart.com   
151929  2.021020e+13     couriermail.com.au   

                                                        4    5  \
0                    https://tass.com/emergencies/1250941  NaN   
1       https://observer-reporter.com/publications/shl...  NaN   
2       http://jordantimes.com/news/local/ujs-css-tops...  NaN   
3       https://apnews.com/1ce7ac194ff97c882f2fe79eebb...  NaN   
4       https://www.wdiy.org/post/hilton-valentine-fou...  NaN   
...                                                   ...  ...   
151925  https://whiznews.com/2021/02/01/liverpool-fill...  NaN   
151926  https://trinidadexpress.com/opinion/letters/la...  NaN   
151927  https://wfmf.iheart.com/content/2021-02-01-mil...  NaN   
151928  https://magic1079.iheart.com/content/2021-02-0...  NaN   
151929  https://www.couriermail.com.au/entertainment/t...  NaN   

                                                        7  \
0       TAX_FNCACT;TAX_FNCACT_CHILDREN;ARREST;CRISISLE...   
1       GENERAL_HEALTH;HEALTH_PANDEMIC;WB_635_PUBLIC_H...   
2       EDUCATION;SOC_POINTSOFINTEREST;SOC_POINTSOFINT...   
3       TAX_DISEASE;TAX_DISEASE_CANCER;WB_1406_DISEASE...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY;TAX_ETHNICITY_ASHANTI;KILL;MANMA...   
151927  TAX_FNCACT;TAX_FNCACT_SINGER;TAX_ETHNICITY;TAX...   
151928  PUBLIC_TRANSPORT;TRIAL;TAX_WORLDFISH;TAX_WORLD...   
151929  TAX_ECON_PRICE;MEDIA_SOCIAL;TAX_FNCACT;TAX_FNC...   

                                                        8  \
0       SOC_POINTSOFINTEREST_POLICE_STATIONS,314;TAX_F...   
1       MANMADE_DISASTER_IMPLIED,1548;MANMADE_DISASTER...   
2       TAX_FNCACT_DIRECTOR,1343;USPEC_POLICY1,769;USP...   
3       MANMADE_DISASTER_RADIOACTIVE,975;ARMEDCONFLICT...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY_ASHANTI,12;MANMADE_DISASTER_IMPL...   
151927  MEDIA_SOCIAL,891;TAX_FNCACT_SINGER,98;TAX_FNCA...   
151928  TRIAL,77;TAX_WORLDFISH_FISH,523;UNGP_FORESTS_R...   
151929  MEDIA_MSM,1795;MEDIA_SOCIAL,594;TAX_ECON_PRICE...   

                                                        9  \
0       4#Moscow, Moskva, Russia#RS#RS48#55.7522#37.61...   
1                  1#Lebanon#LE#LE#33.833333#35.833333#LE   
2       4#Amman, (Jo11), Jordan#JO#JO11#31.95#35.9333#...   
3       2#California, United States#US#USCA#36.17#-119...   
4       4#North Shields, North Tyneside, United Kingdo...   
...                                                   ...   
151925  1#Ecuador#EC#EC#-2#-77.5#EC;1#Japan#JA#JA#36#1...   
151926                                                NaN   
151927                                                NaN   
151928                                                NaN   
151929                                                NaN   

                                                       10  \
0       4#Moscow, Moskva, Russia#RS#RS48#25106#55.7522...   
1            1#Lebanon#LE#LE##33.833333#35.833333#LE#2280   
2       1#Jordan#JO#JO##31#36#JO#13;1#Jordan#JO#JO##31...   
3       3#Los Angeles, California, United States#US#US...   
4       5#Northumberland, Northumberland, United Kingd...   
...                                                   ...   
151925  1#Japan#JA#JA##36#138#JA#21

In [12]:
df['1'] = [int(float(i)) for i in df['1']]
df

1                      3  \
0       20210201000000               tass.com   
1       20210201000000  observer-reporter.com   
2       20210201000000        jordantimes.com   
3       20210201000000             apnews.com   
4       20210201000000               wdiy.org   
...                ...                    ...   
151925  20210201234500           whiznews.com   
151926  20210201234500    trinidadexpress.com   
151927  20210201234500             iheart.com   
151928  20210201234500             iheart.com   
151929  20210201234500     couriermail.com.au   

                                                        4    5  \
0                    https://tass.com/emergencies/1250941  NaN   
1       https://observer-reporter.com/publications/shl...  NaN   
2       http://jordantimes.com/news/local/ujs-css-tops...  NaN   
3       https://apnews.com/1ce7ac194ff97c882f2fe79eebb...  NaN   
4       https://www.wdiy.org/post/hilton-valentine-fou...  NaN   
...                                                   ...  ...   
151925  https://whiznews.com/2021/02/01/liverpool-fill...  NaN   
151926  https://trinidadexpress.com/opinion/letters/la...  NaN   
151927  https://wfmf.iheart.com/content/2021-02-01-mil...  NaN   
151928  https://magic1079.iheart.com/content/2021-02-0...  NaN   
151929  https://www.couriermail.com.au/entertainment/t...  NaN   

                                                        7  \
0       TAX_FNCACT;TAX_FNCACT_CHILDREN;ARREST;CRISISLE...   
1       GENERAL_HEALTH;HEALTH_PANDEMIC;WB_635_PUBLIC_H...   
2       EDUCATION;SOC_POINTSOFINTEREST;SOC_POINTSOFINT...   
3       TAX_DISEASE;TAX_DISEASE_CANCER;WB_1406_DISEASE...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY;TAX_ETHNICITY_ASHANTI;KILL;MANMA...   
151927  TAX_FNCACT;TAX_FNCACT_SINGER;TAX_ETHNICITY;TAX...   
151928  PUBLIC_TRANSPORT;TRIAL;TAX_WORLDFISH;TAX_WORLD...   
151929  TAX_ECON_PRICE;MEDIA_SOCIAL;TAX_FNCACT;TAX_FNC...   

                                                        8  \
0       SOC_POINTSOFINTEREST_POLICE_STATIONS,314;TAX_F...   
1       MANMADE_DISASTER_IMPLIED,1548;MANMADE_DISASTER...   
2       TAX_FNCACT_DIRECTOR,1343;USPEC_POLICY1,769;USP...   
3       MANMADE_DISASTER_RADIOACTIVE,975;ARMEDCONFLICT...   
4                                                     NaN   
...                                                   ...   
151925                                                NaN   
151926  TAX_ETHNICITY_ASHANTI,12;MANMADE_DISASTER_IMPL...   
151927  MEDIA_SOCIAL,891;TAX_FNCACT_SINGER,98;TAX_FNCA...   
151928  TRIAL,77;TAX_WORLDFISH_FISH,523;UNGP_FORESTS_R...   
151929  MEDIA_MSM,1795;MEDIA_SOCIAL,594;TAX_ECON_PRICE...   

                                                        9  \
0       4#Moscow, Moskva, Russia#RS#RS48#55.7522#37.61...   
1                  1#Lebanon#LE#LE#33.833333#35.833333#LE   
2       4#Amman, (Jo11), Jordan#JO#JO11#31.95#35.9333#...   
3       2#California, United States#US#USCA#36.17#-119...   
4       4#North Shields, North Tyneside, United Kingdo...   
...                                                   ...   
151925  1#Ecuador#EC#EC#-2#-77.5#EC;1#Japan#JA#JA#36#1...   
151926                                                NaN   
151927                                                NaN   
151928                                                NaN   
151929                                                NaN   

                                                       10  \
0       4#Moscow, Moskva, Russia#RS#RS48#25106#55.7522...   
1            1#Lebanon#LE#LE##33.833333#35.833333#LE#2280   
2       1#Jordan#JO#JO##31#36#JO#13;1#Jordan#JO#JO##31...   
3       3#Los Angeles, California, United States#US#US...   
4       5#Northumberland, Northumberland, United Kingd...   
...                                                   ...   
151925  1#Jap

In [20]:
df = df[(df['3']=='yahoo.com') | (df['3']=='reuters.com') | (df['3']=='marketwatch.com') | (df['3']=='prnewswire.com')]
df

1                3  \
47      20210201000000      reuters.com   
70      20210201000000      reuters.com   
318     20210201000000        yahoo.com   
367     20210201000000        yahoo.com   
370     20210201000000      reuters.com   
...                ...              ...   
151796  20210201234500  marketwatch.com   
151827  20210201234500      reuters.com   
151853  20210201234500      reuters.com   
151881  20210201234500  marketwatch.com   
151904  20210201234500   prnewswire.com   

                                                        4  \
47      https://www.reuters.com/article/us-myanmar-pol...   
70      https://www.reuters.com/article/us-health-coro...   
318     https://news.yahoo.com/republican-adam-kinzing...   
367     https://news.yahoo.com/usc-could-uclas-jalen-h...   
370     https://www.reuters.com/article/us-retail-trad...   
...                                                   ...   
151796  https://www.marketwatch.com/press-release/dubb...   
151827  https://www.reuters.com/article/us-usa-white-s...   
151853  https://www.reuters.com/article/us-mediatek-5g...   
151881  https://www.marketwatch.com/story/biden-meets-...   
151904  https://www.prnewswire.com/news-releases/ironc...   

                                                        5  \
47                                                    NaN   
70      KILL#19#means#2#New York, United States#US#USN...   
318     KILL#5##2#Illinois, United States#US#USIL#40.3...   
367                                                   NaN   
370                                                   NaN   
...                                                   ...   
151796                                                NaN   
151827                                                NaN   
151853                                                NaN   
151881  KILL#450#nears#1#United States#US#US#39.828175...   
151904                                                NaN   

                                                        7  \
47      TAX_FNCACT;TAX_FNCACT_LEADER;USPEC_POLITICS_GE...   
70      TAX_ETHNICITY;TAX_ETHNICITY_BLACK;HEALTH_VACCI...   
318     LEADER;TAX_FNCACT;TAX_FNCACT_CONGRESSMAN;USPEC...   
367     SOC_EMERGINGTECH;TAX_FNCACT;TAX_FNCACT_EXPLORE...   
370     ENV_METALS;WB_2937_SILVER;WB_507_ENERGY_AND_EX...   
...                                                   ...   
151796  CRISISLEX_C07_SAFETY;UNGP_FORESTS_RIVERS_OCEAN...   
151827  TAX_FNCACT;TAX_FNCACT_REPRESENTATIVE;LEADER;TA...   
151853  TAX_ETHNICITY;TAX_ETHNICITY_TAIWANESE;TAX_FNCA...   
151881  LEADER;TAX_FNCACT;TAX_FNCACT_PRESIDENT;USPEC_P...   
151904  WB_678_DIGITAL_GOVERNMENT;WB_652_ICT_APPLICATI...   

                                                        8  \
47      IDEOLOGY,1198;GENERAL_GOVERNMENT,322;EPU_POLIC...   
70      IDEOLOGY,2113;MEDIA_MSM,733;HEALTH_VACCINATION...   
318     TAX_POLITICAL_PARTY_REPUBLICANS,148;TAX_POLITI...   
367     TAX_FNCACT_NAVIGATOR,3847;TAX_FNCACT_NAVIGATOR...   
370     IDEOLOGY,583;TAX_ECON_PRICE,65;TAX_FNCACT_MINE...   
...                                                   ...   
151796  WB_652_ICT_APPLICATIONS,3242;WB_660_BUSINESS_I...   
151827  IDEOLOGY,1833;WB_2467_TERRORISM,1219;WB_2433_C...   
151853  IDEOLOGY,1607;TAX_FNCACT_DESIGNER,66;TAX_FNCAC...   
151881  RURAL,3742;TAX_POLITICAL_PARTY_REPUBLICANS,109...   
151904  TAX_FNCACT_MANAGER,2836;WB_135_TRANSPORT,206;W...   

                                                        9  \
47      4#London, London, City Of, United Kingdom#UK#U...   
70      2#New York, United States#US#USNY#42.1497#-74....   
318     2#Florida, United States#US#USFL#27.8333#-81.7...   
367                                                   NaN   
370     1#Australia#AS#AS#-25#135#AS;1#Singapore#SN#SN...   
...                                                   ...   
151796  1#Australia#AS#AS#-25#135#AS;4#Melbourne, Vict...   
151827  1#United States#US#US#39.828175#-98.5795#US;3#...   
151853  1#China#CH#CH#3

# EDA

## Generate website counts to determine how to filter 

In [ ]:
websites = df.loc[:,'3']
#print(websites)

In [ ]:
# 50 most popular websites
websites.str.split(";",expand=True).stack().value_counts()[0:100]

In [ ]:
websites_df = websites.str.split(";",expand=True).stack().value_counts()[0:100].rename_axis('website').reset_index(name='counts')
websites_df

In [ ]:
websites_df.to_excel("website.xlsx")

iheart.com                 32071
msn.com                    16704
medium.com                  9739
reuters.com                 6849
indiatimes.com              5370
yahoo.com                   5367
prnewswire.com              4368
dailymail.co.uk             3807
apnews.com                  3542
texasguardian.com           3450
sandiegosun.com             3117
newyorktelegraph.com        3043
ianslive.in                 2719
marketwatch.com             2647
washingtontimes.com         2461
sfgate.com                  2452
chron.com                   2449
onenewspage.com             2256
tmcnet.com                  2250
thenews.com.pk              2157
investegate.co.uk           2137
freerepublic.com            1997
thehindu.com                1976
upstract.com                1935
business-standard.com       1793
lmtonline.com               1760
finanznachrichten.de        1749
express.co.uk               1729
cnet.com                    1720
chinadaily.com.cn           1697
prokerala.com               1575
bignewsnetwork.com          1573
thestar.com.my              1532
siouxcityjournal.com        1515
aninews.in                  1508
allafrica.com               1478
moneycontrol.com            1443
mirror.co.uk                1427
thetimes.co.uk              1336
patch.com                   1336
daijiworld.com              1330
ghanaweb.com                1318
timesnownews.com            1225
washingtonpost.com          1219
news18.com                  1211
princegeorgecitizen.com     1210
xinhuanet.com               1207
digitaljournal.com          1202
openpr.com                  1194
nzherald.co.nz              1193

## Generate keyword count

In [ ]:
keywords = df.loc[:,'5']
#print(keywords)

In [ ]:
# 100 most popular keywords
keywords_df = keywords.str.split("#",expand=True).stack().value_counts()[0:100].rename_axis('keywords').reset_index(name='counts')
keywords_df

In [ ]:
keywords_df.to_excel("keyword.xlsx")

## Generate topics count

In [ ]:
topics = df.loc[:,'7']
#print(topics)

In [ ]:
# 100 most populare topics
topics_df = topics.str.split(";",expand=True).stack().value_counts()[0:100].rename_axis('topics').reset_index(name='counts')
topics_df

In [ ]:
topics_df.to_excel("topic.xlsx")

## Convert datetime stamp

In [ ]:
date_time = df.loc[:,'1']
print(type(dt.strptime(date_time[len(date_time)], '%Y%m%d%H%M%S')))
print(dt.strptime(date_time[len(date_time)], '%Y%m%d%H%M%S'))

In [ ]:
df['1'] =[dt.strptime(str(d), '%Y%m%d%H%M%S') for d in date_time]
df